### PR TITLE
Android NDK detection adjust

### PR DESCRIPTION
**Pull request by madrazo**
-Use getNdkVersion to check version in both source.properties file and
name of directory.  It returns a float value that can be compared.
-Try to use the default ndk-bundle from SDK manager if there is no valid
NDK in ANDROID_NDK_DIR
-fix NDKVXX flag to force looking for NDKs even if ANDROID_NDK_ROOT is
set

### CODE REVIEW

File: tools/hxcpp/Setup.hx [hunk 1]
```diff
@@ -5,82 +5,83 @@ import BuildTool;
...
+        try
+        {
+             files = FileSystem.readDirectory(ndkDir);
+        }
+        catch (e:Dynamic)
+        {
+             Log.warn('ANDROID_NDK_DIR "$ndkDir" does not point to a valid directory');
+             checkFiles=false;
+        }
+        if (checkFiles)
```
Comment (thomasuster) [inline tools/hxcpp/Setup.hx]: Is this bool needed anymore? Seems like you could just call a function to check the files inside of the catch.

Comment (madrazo) [reply tools/hxcpp/Setup.hx]: Initialy this code only considered the name of directories to detect the NDK version, and later added to check source.properties file.  The check is done in 2 places, but now it calls getNdkVersion on both. Things start to be a little tricky when trying to use the NDKVXX flag. I believe this code fixes it when retrieving the version from the source.properties file and also gives a higher prority than ANDROID_NDK_ROOT. It also checks in the default Android studio ndk-bundle directory.

Comment (madrazo) [reply tools/hxcpp/Setup.hx]: Yes, maybe it would be the same as writting the checkFiles block inside the "try"

**Comment by Gama11** [comment]
These changes cause Android compilation to break for me, see #715.

**Comment by madrazo** [comment]
Something that is different:
var extract_version = ~/^(android-ndk-)?r(\d+)([a-z]?)$/;

var extract_version = ~/.*\/(android-ndk-)?r(\d+)/;    

Please give it a try

**Pull request merged**